README.md

```diff
@@ -1,33 +1,164 @@
+# api-client-pkg
+
+[![CircleCI](https://circleci.com/gh/smg-automotive/api-client-pkg/tree/main.svg?style=svg&circle-token=c183f151fea3c74453cf8dd962d31e115906a300)](https://circleci.com/gh/smg-automotive/example-pkg/tree/main)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
 ## Usage
+
 ```
-npm install @smg-automotive/example
+npm install @smg-automotive/api-client-pkg
 ```
 
+````typescript
+import { ApiClient } from "@smg-automotive/api-client-pkg"
+````
+
+### Configuration
+
+The available instance methods are listed below. The specified config will be merged with the instance config.
+
+````typescript
+ApiClient.configure({
+  baseUrl: 'https://api.automotive.ch/api',
+  headers: { 'Content-Type': 'application/json' },
+})
+````
+
+### Instance methods
+
+The ApiClient provides promise-based methods for `GET`, `POST`, `PUT` and `DELETE`.
+
+### Typing
+
+You can define the request and the response type of your method as follows:
+
+````typescript
+interface Car {
+  make: string
+  horsepower: number
+}
+
+// data has type Car
+var data = await ApiClient.get<Car>('/listings/search')
+
+// data has type { id: string } and body is typed as Car
+var data = await ApiClient.post<{ id: string }, Car>('/listings/create',
+  { make: 'bmw', horsepower: 300 },)
+````
+
+### Parameters
+
+If your request URL contains parameters, you can use curly brackets and the ApiClient replaces it with the passed data.
+This is aligned with the OpenAPI specification used by Swagger. Hence, you can easily search for occurrences of your
+APIs.
+
+````typescript
+// fetch is called with https://baseUrl.api.ch/dealers/123/listings/456
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
+})
+````
+
+### Authorization
+
+You can pass `accessToken` to the request options if you are talking to a protected API. The client will automatically
+set the header with a Bearer token.
+
+````typescript
+// fetch is called with the header { Authorization: `Bearer ${accessToken}` }
+await ApiClient.post<{ id: number }, { name: string; listingIds: number[] }>(
+  "users/me/listing-comparisons",
+  { name: "test1", listingIds: [890163] },
+  { accessToken: authHeader.accessToken }
+)
+````
+
+### Error handling
+
+The promise is rejected if any error occurs on API level. Use `try/catch` to add error handling.
+
+````typescript
+try {
+  // res has the type LeasingCalculation
+  const res = await ApiClient.post<LeasingCalculation, LeasingData>(
+    "listings/calculate-leasing",
+    {
+      downPayment: 7300,
+      duration: 48,
+      estimatedKmPerYear: 10000,
+      firstRegistrationDate: "2020-07-01",
+      residualValue: 12045,
+      price: 36500,
+    }
+  )
+} catch (error: ResponseError) {
```

Comment:
an error in the api client does not necessarily need to be a ResponseError. I would discourage including such a confusing example in the README

also, where is `ResponseError` defined?

Reply:
every error coming from the API is a response error. Other errors can be wrong parameters or fetch itself fails. I removed it from the example but 99% of the cases it is a response error. 
ResponseError is defined in the api client as a [subclass](https://github.com/smg-automotive/api-client-pkg/pull/1/files#diff-9b21498755dfb2899618ba6439333ba17cb717e891766b6d39bb23c5b2f0ee54) of Error.